src/app/components/TradingChart/index.tsx

```diff
@@ -0,0 +1,174 @@
+/**
+ *
+ * TradingChart
+ *
+ */
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import Chart from 'utils/kaktana-react-lightweight-charts/Chart';
+import { Skeleton } from '../PageSkeleton';
+import { backendUrl, currentChainId } from 'utils/classifiers';
+
+enum Theme {
+  LIGHT = 'Light',
+  DARK = 'Dark',
+}
+
+enum ChartType {
+  LINE = 'line',
+  CANDLE = 'candle',
+}
+
+export interface ChartContainerProps {
+  rate: number;
+  symbol: string;
+  type: ChartType;
+  theme: Theme;
+}
+
+// detailed description of series options https://github.com/kaktana/kaktana-react-lightweight-charts#seriesobject
+interface ChartData {
+  data: any[];
+  options?: any;
+  markers?: any;
+  priceLines?: any;
+  legend?: string;
+  linearInterpolation?: number;
+}
+
+export function TradingChart(props: ChartContainerProps) {
+  const threeMonths = 7257600000; //3 months in ms
+  const initData: ChartData[] = [{ data: [] }];
+  const [hasCharts, setHasCharts] = useState<boolean>(false);
+  const [chartData, setChartData] = useState<ChartData[]>(initData.slice());
+  const [lastTime, setLastTime] = useState<number>(
+    new Date().getTime() - threeMonths,
+  );
+
+  const resetChart = () => {
+    setHasCharts(false);
+    setChartData(initData.slice());
+    setLastTime(new Date().getTime() - threeMonths);
+  };
+
+  const seriesProps = {
+    candlestickSeries: props.type === ChartType.CANDLE ? chartData : undefined,
+    lineSeries: props.type === ChartType.LINE ? chartData : undefined,
+  };
+
+  useEffect(() => {
+    function getData() {
+      axios
+        .get(`${backendUrl[currentChainId]}/datafeed/price/${props.symbol}`, {
+          params: {
+            type: props.type,
+            startTime: lastTime,
+          },
+        })
+        .then(response => {
+          //console.log(response);
+          if (response.data && response.data.series) {
+            let newSeries: Array<any> = [];
+            if (chartData[0].data.length > 0) {
+              // remove old datums, starting at the one with same time as first new one that was received
+              newSeries = chartData[0].data.slice();
+              const firstVal = response.data.series[0];
+              const len = newSeries.length - 1;
+              let i = -1;
+              for (let x = len; x >= 0; x--)
+                if (newSeries[x].time === firstVal.time) {
+                  i = x;
+                  break;
+                }
+              if (i > -1) newSeries = newSeries.slice(0, i);
+            }
+            newSeries = newSeries.concat(response.data.series);
+            // console.log(newSeries);
+            setChartData([{ data: newSeries }]);
+            const latest = newSeries[newSeries.length - 1];
+            setLastTime(latest.time * 1e3); // datum time is in seconds, lastTime is ms
```

Comment:
When backend returns empty `series` array chart fails to load and console shows this error `TypeError: Cannot read property 'time' of undefined`

Checking if response.data.series.length is not 0 here too should do the trick